basculehttp/tokenFactory.go

```diff
@@ -74,6 +75,46 @@ func (btf BasicTokenFactory) ParseAndValidate(ctx context.Context, _ *http.Reque
 	return bascule.NewToken("basic", principal, bascule.NewAttributes()), nil
 }
 
+// NewBasicTokenFactoryFromMap takes a map of usernames to passwords and returns
+// the BasicTokenFactory with those values.
+func NewBasicTokenFactoryFromMap(m map[string]string) (BasicTokenFactory, error) {
```

Comment:
My bad for suggesting this:  I didn't realize `BasicTokenFactory` is itself an alias for `map[string]string`.  That makes this function superfluous.  Client code can just do `bascule.BasicTokenFactory(m)` to turn a map into a factory.